Basestation_Software.Web/Core/Components/RamanGraph.razor

```diff
@@ -0,0 +1,146 @@
+﻿@inject RoveCommService _RoveCommService
+
+<div class="card full-height">
+   <div class="card-header">
+       <h5>Raman Spectroscopy Graph</h5>
+   </div>
+   <p>
+       <RadzenChart>
+          <RadzenLineSeries Data="@ramanGraph" CategoryProperty="Raman_Shift" ValueProperty="Intensity" />
+       </RadzenChart>
+   </p>
+   <button @onclick=RequestRamanData> Request Raman Spectroscopy Data </button>
+</div> 
+
+
+@code {
+    //Variables
+    private static List<ushort> _raman1 = new List<ushort>(500),
+    _raman2 = new List<ushort>(500),
+    _raman3 = new List<ushort>(500),
+    _raman4 = new List<ushort>(500),
+    _raman5 = new List<ushort>(48);
+
+
+    class DataItem
+    {
+        public int Raman_Shift { get; set; }
+        public double Intensity { get; set; }
+
+        public DataItem()
+        {
+            Raman_Shift = 0;
+            Intensity = 0f;
+        }
+    }
+
+    DataItem[] ramanGraph = new DataItem[2048];
+
+
+    private async void RequestRamanData()
+    {
+        await _RoveCommService.SendAsync<uint>("Instruments", "RequestRamanReading", [0], reliable: false);
+
+        for (int i = 0; i < 2048; i++)
+        {
+            if (i < 500)
+            {
+                ramanGraph[i] = new DataItem
+                {
+                    Raman_Shift = i,
+                    Intensity = _raman1[i],
```

Comment:
If I am understanding this properly, this might be a potential race condition. The await _RoveCommService will return as soon as it has sent the packet, but it doesn't wait for the data to arrive and the callback to be triggered before returning. So you it send the packet and then read the data before it has been received or while it's being received, which is not good. Maybe move these if statements to the callbacks or setup a delegate and event that will trigger when the callbacks are done receiving data.